### PR TITLE
Updating AssetBrowser tests around new TableView model

### PR DESCRIPTION
**Pull request by jckand-amzn**
Signed-off-by: jckand-amzn <jckand@amazon.com>

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/editor/EditorScripts/AssetBrowser_SearchFiltering.py [hunk 1]
```diff
@@ -94,16 +94,21 @@ def verify_files_appeared(model, allowed_asset_extentions, parent_index=QtCore.Q
...
         editor_window = pyside_utils.get_editor_main_window()
         app = QtWidgets.QApplication.instance()
-        
-        # 3) Type the name of an asset in the search bar and make sure only one asset is filtered in Asset browser
+
+        # 3) Type the name of an asset in the search bar and make sure it is filtered to and selectable
         asset_browser = editor_window.findChild(QtWidgets.QDockWidget, "Asset Browser")
         search_bar = asset_browser.findChild(QtWidgets.QLineEdit, "textSearch")
         search_bar.setText("cedar.fbx")
         asset_browser_tree = asset_browser.findChild(QtWidgets.QTreeView, "m_assetBrowserTreeViewWidget")
```
Comment (jackalbe) [inline AutomatedTesting/Gem/PythonTests/editor/EditorScripts/AssetBrowser_SearchFiltering.py:102]: why findChild() asset_browser_tree if asset_browser_tree is not used?

**Review by jackalbe — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/editor/EditorScripts/AssetBrowser_SearchFiltering.py (hunk 1 above)
Comment (jckand-amzn) [reply AutomatedTesting/Gem/PythonTests/editor/EditorScripts/AssetBrowser_SearchFiltering.py:102]: It's used further down for filtering checks outside of the search context.

**Review by AMZN-Igarri — APPROVED** [review]

**Comment by hultonha** [comment]
@AMZN-Igarri I'm just curious as the Asset Browser code might well change again in future, we should definitely check these tests when making changes to ensure we don't break them next time, and can maybe make them more resistant to these kinds of changes (Asset Picker too)

**Comment by jackalbe** [comment]
> the Asset Browser code might well change again in future,
is there a road map showing this? There are a number of automated tests that need to be upgrade so that when the changes come in the team making the changes to the AB will update those tests as well

**Comment by jckand-amzn** [comment]
> 
> 
> @AMZN-Igarri I'm just curious as the Asset Browser code might well change again in future, we should definitely check these tests when making changes to ensure we don't break them next time, and can maybe make them more resistant to these kinds of changes (Asset Picker too)

Yeah, would have caught the changes immediately if the optimized test suite was enabled to be default. As it sits right now, though these tests were only running without an xfail in the non-optimized periodic suite. I have a task to get everything moved to run as an optimized test by default which will put more tests into the main suite and catch these issues in the future quicker.

**Comment by AMZN-Igarri** [comment]
> 
> 
> > the Asset Browser code might well change again in future,
> 
> is there a road map showing this? There are a number of automated tests that need to be upgrade so that when the changes come in the team making the changes to the AB will update those tests as well

Yes I agree. If this sound good we could have a chat about the Asset Browser tests so we can avoid this kind of issues in the future @jackalbe @jckand-amzn

**Review by AMZN-daimini — APPROVED** [review]

**Pull request merged**